server.go

```diff
@@ -376,6 +381,15 @@ func (s *server) sendToPeer(target *btcec.PublicKey, msgs ...lnwire.Message) err
 	return nil
 }
 
+// findPeer will return the peer that corresponds to the passed in public key.
+// This function is used by the funding manager, allowing it to update the
+// daemon's local representation of the remote peer.
+func (s *server) findPeer(peerKey *btcec.PublicKey) *peer {
+	serializedIdKey := string(peerKey.SerializeCompressed())
+
+	return s.peersByPub[serializedIdKey]
```

Comment:
This map access should ensure that it has the read-lock of the `peersMtx` before accessing it. This current version isn't thread-safe.